Subgraph updates to development

codenamejason opened the pull request:
Jason and Daniele working branch to update development

tim-schultz committed:
wip github auth

codenamejason committed:
[WIP] adding WAGMI provider

codenamejason committed:
fixed window.ethereum error

codenamejason committed:
mock up base modal to test buttons

codenamejason committed:
fix close modal

codenamejason committed:
grabbing state

DanieleSalatti committed:
WIP modal work

DanieleSalatti committed:
WIP modal work

codenamejason committed:
Merge pull request #205 from gitcoinco/main

update branch 183-wallet-connect with latest main

codenamejason committed:
add account hook and test connection

tim-schultz committed:
removed unneeded libraries

tim-schultz committed:
making requests to receive credential

codenamejason committed:
learning how to use redux lol

codenamejason committed:
Merge pull request #207 from gitcoinco/main

rebase for 183-wallet-connect

codenamejason committed:
make some more progress

codenamejason committed:
share the components from RM for wallet connect

codenamejason committed:
a bit of progress

codenamejason committed:
address not showing in nav

tim-schultz committed:
added icons for verification flow

tim-schultz committed:
colors for verification flow

tim-schultz committed:
remove oauth listener from layout

tim-schultz committed:
additional button variant

tim-schultz committed:
emit verification status to parent

tim-schultz committed:
add project verifiation step

tim-schultz committed:
hook up project verification step

tim-schultz committed:
move callback stream to landing page

tim-schultz committed:
unify styling

tim-schultz committed:
added error state to verification step

tim-schultz committed:
removed unneeded verification functions

codenamejason committed:
just about done, going to test some more

DanieleSalatti committed:
Add subgraph

DanieleSalatti committed:
Add subgraph

DanieleSalatti committed:
Add subgraph

tim-schultz committed:
resolve issues with remaining tests

tim-schultz committed:
linting

tim-schultz committed:
fixing remaining linting errors

tim-schultz committed:
remove placeholder input

tim-schultz committed:
added verification error

codenamejason committed:
everything seems like it's working 🫣

codenamejason committed:
some clean up and linting..

codenamejason committed:
almost done linting

codenamejason committed:
latest linting

DanieleSalatti committed:
Add graphqlClient and queries

DanieleSalatti committed:
Add graphqlClient and queries

DanieleSalatti committed:
Fetch projects from subgraph

DanieleSalatti committed:
Fetch projects from subgraph

DanieleSalatti committed:
Merge branch 'feature/subgraph' of github.com:DanieleSalatti/grant-hub into feature/subgraph

DanieleSalatti committed:
Correctly parse round data

codenamejason committed:
smthng going on with reload on with network switch

codenamejason committed:
quick update for network switch flow

codenamejason committed:
add current network name to header

DanieleSalatti committed:
Add useLocalStorage hook

DanieleSalatti committed:
Add useLocalStorage hook

DanieleSalatti committed:
WIP application modal and flow

DanieleSalatti committed:
WIP application modal and flow

codenamejason committed:
Merge branch 'main' into 183-wallet-connect

codenamejason committed:
fix build error from rebase

tim-schultz committed:
updated provider type

DanieleSalatti committed:
Merge branch 'feature/subgraph' of github.com:DanieleSalatti/grant-hub into feature/subgraph

DanieleSalatti committed:
Merge branch 'feature/111-113-modals' of github.com:DanieleSalatti/grant-hub into feature/111-113-modals

codenamejason committed:
Merge pull request #220 from gitcoinco/main

update working branch

codenamejason committed:
before merge with DS wip branch

tim-schultz committed:
Merge branch 'main' of github.com:gitcoinco/grant-hub into 193-github-oauth

Commit:
192/193 - Github and Twitter oauth (#219)

* wip adding twitter 0auth

* listen for twitter redirect

* update twitter type

* emit error message from provider

* add twitter and github verification

* twitter -> github

* remove log

* save twitter VC to state

* added datadog error logging to providers

tim-schultz committed:
remove unnecesary types

tim-schultz committed:
debounce on message callback

tim-schultz committed:
add verification form to edit flow

DanieleSalatti committed:
Merge branch '183-wallet-connect' of https://github.com/gitcoinco/grant-hub into feature/111-113-modals

DanieleSalatti committed:
Fux merge issues

DanieleSalatti committed:
Merge pull request #231 from DanieleSalatti/183-wallet-connect

Working session - delete later - do not review

codenamejason committed:
add null check

tim-schultz committed:
added projectForm state

tim-schultz committed:
save form values to store

codenamejason committed:
update gql client with wagmi

codenamejason committed:
update actions to use wagmi

codenamejason committed:
some more wagmi updates

codenamejason committed:
some clean up

tim-schultz committed:
saving credentials to new rpoject store

tim-schultz committed:
save project metadata

codenamejason committed:
add rainbow kit, working through error

tim-schultz committed:
project detail component

tim-schultz committed:
use project detail component

tim-schultz committed:
use ProjectFormStatus enum to track status

tim-schultz committed:
moved veriifcation form

tim-schultz committed:
save credentials and move to next step

DanieleSalatti committed:
Fix subgraph query

DanieleSalatti committed:
Merge branch '183-wallet-connect' of https://github.com/gitcoinco/grant-hub into 183-wallet-connect

codenamejason committed:
update sample.env

tim-schultz committed:
preview component

tim-schultz committed:
allow details to render Blob for preview

tim-schultz committed:
use preview comopnent

codenamejason committed:
update wagmi client and clean up

codenamejason committed:
testing

tim-schultz committed:
submit project from preview page

tim-schultz committed:
seperate projectForm state from creation call

tim-schultz committed:
navigate to projects page after creation

tim-schultz committed:
use store to track form values

tim-schultz committed:
reset form values on successful submission

tim-schultz committed:
remove form values from store on exit

tim-schultz committed:
load edit project if it exists

tim-schultz committed:
load existing img by type

tim-schultz committed:
var rename

tim-schultz committed:
add preview step to edit flow

tim-schultz committed:
standardize image format on edit for saving to IPFS

tim-schultz committed:
pass existing id on edit form

tim-schultz committed:
maintain existing styles on Show

tim-schultz committed:
save credentials to project

tim-schultz committed:
show credential on detail

codenamejason committed:
Merge branch '183-wallet-connect' into 193-github-oauth

codenamejason committed:
Merge pull request #236 from gitcoinco/193-github-oauth

193 GitHub oauth to update 183

codenamejason committed:
Merge branch 'gitcoinco:183-wallet-connect' into 183-wallet-connect

codenamejason committed:
fix build error

tim-schultz committed:
show credential on detail

DanieleSalatti committed:
WIP switch to subgraph

codenamejason committed:
need some tailwind assist.

DanieleSalatti committed:
Fix new grant form not showing

codenamejason committed:
fix landing page from Figma

tim-schultz committed:
display credentials on preview

tim-schultz committed:
remove log

tim-schultz committed:
removed unused methods and tests, same methods are found in passport verifier

tim-schultz committed:
added remaining env variables to client

codenamejason committed:
Merge branch '183-wallet-connect' into 195-save-verified-credentials

codenamejason committed:
Merge pull request #241 from gitcoinco/195-save-verified-credentials

195 save verified credentials to update 183

codenamejason committed:
Merge branch 'gitcoinco:183-wallet-connect' into 183-wallet-connect

codenamejason committed:
Merge pull request #242 from codenamejason/183-wallet-connect

183 wallet connect

codenamejason committed:
some lint fixes and style updates, wallet nds work.

DanieleSalatti committed:
Merge commit '94cd6a9e72e1df27d63cccddbb9a85f3215bcdda' into 183-wallet-connect

DanieleSalatti committed:
Merge commit 'c1ce9773bb2367d13ed9c736a27ac17efd9a4d6a' into 183-wallet-connect

DanieleSalatti committed:
New lockfile

codenamejason committed:
works but looks like shit 🤬

codenamejason committed:
start redoing wallet disply with chakra

DanieleSalatti committed:
Merge commit 'febf6c88113b9d58e2a689a180597b30d3809f8e' into 183-wallet-connect

codenamejason committed:
some style updates

codenamejason committed:
now to merge and get updates

codenamejason committed:
update network switch chains

codenamejason committed:
Merge branch 'development' into 183-wallet-connect

codenamejason committed:
fix merge errors

DanieleSalatti committed:
WIP subgraph

DanieleSalatti committed:
project form fixes

DanieleSalatti committed:
Merge branch 'feature/subgraph' into subgraph-updates-jaxcoder

# Conflicts:
#	client/src/actions/newGrant.ts
#	client/src/components/base/WalletDisplay.tsx
#	client/src/components/grants/List.tsx
#	client/src/components/rounds/Apply.tsx
#	client/src/components/rounds/Show.tsx

codenamejason committed:
linting

DanieleSalatti committed:
Merge branch 'subgraph-updates' of github.com:codenamejason/grant-hub into subgraph-updates-jc

# Conflicts:
#	client/src/actions/newGrant.ts

codenamejason committed:
some error fixes

DanieleSalatti committed:
Project creation works again

codenamejason committed:
Merge pull request #2 from DanieleSalatti/subgraph-updates-jaxcoder

Project creation works again

codenamejason committed:
some clean up now we are using wagmi

codenamejason committed:
Merge branch 'subgraph-updates' of https://github.com/codenamejason/grant-hub into subgraph-updates

DanieleSalatti committed:
Apply to round fixes

codenamejason committed:
Merge pull request #3 from DanieleSalatti/subgraph-updates-jaxcoder

Apply to round fixes

codenamejason committed:
fix network switch, working now

codenamejason commented:
This PR also has all the wagmi and rainbow kit updates as well as the subgraph updates

tim-schultz commented on client/src/components/application/Form.tsx:
```diff
@@ -73,8 +74,18 @@ export default function Form({
-      dispatch(submitApplication(round.address, formInputs));
+    if (roundManagerClient && formValidation.valid) {
+      showToast(true);
+      await submitApplication(roundManagerClient, round.address, formInputs)
```
Maybe stick to either `await` or `.then()` here?

DanieleSalatti replied:
Picking this up

tim-schultz commented on client/src/components/base/Preview.tsx:
```diff
@@ -1,75 +1,68 @@
     setSubmitted(true);
     showToast(true);
-    await dispatch(publishGrant(currentProjectId));
+    await publishGrant(formInputs, chain!.id, signer, currentProjectId)
```
Same here with `async` vs `then`  🙏

DanieleSalatti replied:
Picking this up

tim-schultz commented on client/src/components/base/Preview.tsx:
```diff
@@ -1,75 +1,68 @@
 import Toast from "./Toast";
 import TXLoading from "./TXLoading";
-import { ProjectFormStatus } from "../../types";
-import { formReset } from "../../actions/projectForm";
```
Should we delete `actions/projectForm` and `reducers/projectForm` files if it is no longer being used?

DanieleSalatti replied:
Picking this up

tim-schultz commented on client/src/components/base/WalletDisplay.tsx:
```diff
@@ -0,0 +1,143 @@
+          </MenuItem>
+        </MenuList>
+      </Menu>
+      {/* todo: redo this modal using our base setup 
```
Should we wipe these comments out if?

codenamejason replied:
cleaned up

tim-schultz commented on client/src/components/rounds/Show.tsx:
```diff
@@ -1,95 +1,104 @@
+    );
+
+    if (!roundInfo) {
+      console.error("Cannot load round", roundId);
```
@michellema1208 @willsputra Maybe we can get designs for unsuccessful load of round data? Also for loading projects?

willsputra replied:
@schultztimothy @michellema1208 added the error states here!
https://www.figma.com/file/obBzCCL8KwoTlVBb5g3ccN/Grant-Hub---Wireframes?node-id=2189%3A13013

DanieleSalatti replied:
I will address this in a separate PR if that's ok.

codenamejason replied:
added error modal to open on error of round data

tim-schultz commented on client/src/components/rounds/Show.tsx:
```diff
@@ -1,95 +1,104 @@
     }
-  }, [dispatch, roundId]);
-
-  if (props.web3ChainId !== props.roundChainId) {
```
I added this a couple days ago to indicate that the round contract is on a different network than the one the user is connected to https://github.com/gitcoinco/grant-hub/pull/243 Any chance we could add this back in?

DanieleSalatti replied:
Yep, on it!

codenamejason committed:
remove some commented code not needed

codenamejason committed:
add blockie for account

codenamejason committed:
update user icon, blockie not working-removed

codenamejason committed:
some clean up

DanieleSalatti committed:
Fix copy

DanieleSalatti committed:
Merge branch 'subgraph-updates' of github.com:codenamejason/grant-hub into subgraph-updates-jaxcoder

DanieleSalatti committed:
Store round info with chain id and use chain id to fetch round info

DanieleSalatti committed:
Poll the graph and IPFS every few seconds.

This is a temporary solution as The Graph does not support GraphQL subscriptions.

gravityblast commented:
when I create a new project and click on "save and publish", I no longer see the messages in the toast component that were showing "uploading metadata", "uploading images", "waiting to confirm tx", etc, but I see an empty toast:

![Screenshot from 2022-08-16 10-26-37](https://user-images.githubusercontent.com/1059/184834169-69105ef9-6fe2-4e75-912e-190f59c9cff5.png)

gravityblast commented:
I can't apply to the round `0x3879a696253Aa2FaE4164C21e16104894dE632c9` anymore, but we don't show any error, just the loading spinner and a message in the console `Cannot load round 0x3879a696253Aa2FaE4164C21e16104894dE632c9`

![Screenshot from 2022-08-16 10-33-06](https://user-images.githubusercontent.com/1059/184835146-8201f2e4-38bc-4858-8f03-27fb544f05f9.png)

gravityblast commented:
if I go to a round page and then back to the dashboard I see this dialog:

![Screenshot from 2022-08-16 10-52-02](https://user-images.githubusercontent.com/1059/184838981-1756f20e-ed64-4621-868c-c9b9b330fe05.png)

But I think it should be there only when I try to apply to a round and I'm redirected to create a project because I didn't have one yet.
In my case I already had many so I don't think we should show it.

gravityblast commented:
I also noticed it remains there every time I go to the dashboard and if I click to the apply button it redirects to the route without "compiling" the path `#/chains/:chainId/rounds/:roundId/apply`

gravityblast commented:
when I click on kovan or goerli to switch network but then I don't accept, I see an empty page with `Something went wrong.`, maybe we don't catch the error. I would catch the error without doing anything so that the user can stay in the current network.

gravityblast commented on client/src/actions/rounds.ts:
```diff
@@ -75,195 +67,118 @@ const loadingError = (address: string, error: string): RoundsActions => ({
+      roundInfo.round.roundMetaPtr.pointer === undefined ||
+      roundInfo.round.applicationMetaPtr.pointer === undefined
+    ) {
+      // TODO: log error and metric
```
shall we dispatch an error to be able to show it in the UI?

codenamejason replied:
> do we actually need the network switch? it's useful for development but in theory in production we should just stay on the network where the project registry is deployed /cc @michellema1208

I removed it from the header where the chain is displayed.

codenamejason replied:
Also set up env to only allow OP mainnet for prod for OP launch and our test chains for dev env

codenamejason replied:
this was meant for the previous comment...

DanieleSalatti replied:
> shall we dispatch an error to be able to show it in the UI?

This is coming as well

gravityblast commented:
I tried to apply to the latest deployed round at `/#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3``.
With this one I can see the first page, but when I click on "apply to this round" the form doesn't load and I see this error in the console ` TypeError: roundApplication.applicationSchema is undefined`

gravityblast commented:
I see we some calls in the console to both alchemy and infura, are they both needed?

gravityblast commented:
I noticed we have a call to the subgraph to get all the projects, but we also call the subgraph for each project id.
I see the first call returns already everything, shall we remove all the next calls?

gravityblast commented:
do we actually need the network switch? it's useful for development but in theory in production we should just stay on the network where the project registry is deployed /cc @michellema1208

gravityblast reviewed (changes requested):
great job @codenamejason  @DanieleSalatti !
I added some comments on some things we need to fix before merging.

now that all the logic is in the react components and we don't rely anymore on the current state we should 
find the best way to add tests without having to mock all the services we have. 
we should maybe focus on this in the next sprint I think.

great job!!!
:rocket:

codenamejason committed:
autoconnect if previously logged in

DanieleSalatti committed:
Merge commit '7918c4fdd1dabe27755562715d4478f34d036c03' into subgraph-updates-jaxcoder

DanieleSalatti committed:
Merge pull request #4 from DanieleSalatti/subgraph-updates-jaxcoder

Various fixes to the shared branch

codenamejason committed:
turn off network select in header

DanieleSalatti committed:
Address PR feedback

DanieleSalatti committed:
Address PR feedback

DanieleSalatti committed:
Handle roundToApply being null

DanieleSalatti committed:
Handle roundToApply being null

codenamejason committed:
add env for what chains to use per environment

DanieleSalatti committed:
Removing unnecessary calls to subgraph

codenamejason committed:
Merge pull request #5 from DanieleSalatti/subgraph-updates-jaxcoder

Handle roundToApply being null

DanieleSalatti committed:
Merge branch 'subgraph-updates' of github.com:codenamejason/grant-hub into subgraph-updates-jaxcoder

DanieleSalatti commented:
> I also noticed it remains there every time I go to the dashboard and if I click to the apply button it redirects to the route without "compiling" the path `#/chains/:chainId/rounds/:roundId/apply`

This should be fixed in the next update to this PR

codenamejason committed:
remove some commented code

codenamejason committed:
Merge pull request #6 from DanieleSalatti/subgraph-updates-jaxcoder

Subgraph updates jaxcoder

DanieleSalatti commented:
I am adding a 404:

<img width="1315" alt="Schermata 2022-08-16 alle 10 51 38" src="https://user-images.githubusercontent.com/129965/184946039-29e2445b-c2c5-4b42-8368-58285d4f4327.png">

Since we changed the URL for the rounds and I keep going to the old links, which don't handle the error properly.

DanieleSalatti committed:
Add 404

codenamejason committed:
Merge pull request #7 from DanieleSalatti/subgraph-updates-jaxcoder

Add 404

DanieleSalatti committed:
Fix edit grant page

codenamejason committed:
Merge pull request #8 from DanieleSalatti/subgraph-updates-jaxcoder

Fix edit grant page

codenamejason commented:
> > @codenamejason I noticed we no longer "auto-login" users if they already gave permissions to the dapp.
> > We added it in the past as a feature and it's also much better in development to avoid having to login at each refresh :)
> > can we add it back please?
> 
> I notice I'm "logged-out" often even while I make some changes on the code so I would definitely put it back to have development quicker like before ;)

updated to auto login if you have connected previously

codenamejason committed:
add error modal for when round data fetch errors

codenamejason committed:
working on reusable error component

codenamejason committed:
now got to make it look pretty 🧙🏼‍♂️

DanieleSalatti commented:
> I tried to apply to the latest deployed round at `` /#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3``. With this one I can see the first page, but when I click on "apply to this round" the form doesn't load and I see this error in the console `` TypeError: roundApplication.applicationSchema is undefined`

I cannot find that in the graph...

DanieleSalatti committed:
Dispatch error when round metadata not found

DanieleSalatti committed:
Add back check on round network vs. connected network

codenamejason committed:
Merge pull request #9 from DanieleSalatti/subgraph-updates-jaxcoder

Dispatch error when round metadata not found, add back check on round network vs. connected network

codenamejason committed:
Merge branch 'error-modal-updates' into subgraph-updates

codenamejason commented:
I think we got everything @gravityblast and @schultztimothy commented on.

gravityblast commented:
after going to `/#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3b`, and then to the dashboard, I still see the "Time to get your project funded" dialog, which should only showed if I went to the round application without having a project, and I'm redirected to create a new project.

gravityblast commented:
if I go to a non existing round like `/#/chains/5/rounds/0xB91FeC0b68f39cbfdd75E4f08042c60724e1bd3c`, I no longer see an error message but just loading gif and an error in the console

gravityblast commented:
I'm still not able to see the "loading" states we had before when I create a project:

![Uploading Screenshot from 2022-08-16 10-26-37.png…]()

gravityblast commented:
> I cannot find that in the graph...

@DanieleSalatti yeah I guess it was an old one and not indexed, we should put back the error messages to at least say something like `round not found`

codenamejason committed:
addresses round data error

codenamejason committed:
work on the showing of the round modal

schultztimothy commented:
> I'm still not able to see the "loading" states we had before when I create a project:
> 
> ![Screenshot from 2022-08-16 10-26-37](https://user-images.githubusercontent.com/1059/185146472-7275582a-0ab9-427f-bbef-ef50e81a0062.png)

Looks like this one was introduced outside of this PR with the Preview step. I created this to resolve it: https://github.com/gitcoinco/grant-hub/pull/258

codenamejason committed:
update gql query

codenamejason committed:
lint

codenamejason committed:
add the id prop

codenamejason committed:
keep getting null on return of gql query...

codenamejason committed:
update param

DanieleSalatti committed:
Minor fixes to round query

codenamejason committed:
Merge pull request #12 from DanieleSalatti/showing-round-modal-jaxcoder

Minor fixes to round query

codenamejason committed:
setting modal bool up on round apply data

codenamejason committed:
add alert component to apply for op round

DanieleSalatti committed:
Merge pull request #11 from codenamejason/showing-round-modal

work on the showing of the round modal

DanieleSalatti committed:
Applying to a round works again (no progress indication yet)

codenamejason committed:
nuke yarn.lock due to conflicts

codenamejason committed:
small update

DanieleSalatti committed:
New toast working

DanieleSalatti committed:
Accidentally created package.json

DanieleSalatti committed:
Merge branch 'subgraph-updates' of github.com:codenamejason/grant-hub into subgraph-updates

DanieleSalatti committed:
Toast for grant creation

DanieleSalatti committed:
Clean up components, add toast to image loading

DanieleSalatti commented:
Closing in favour of #262, which contains all the changes from this PR in a conveniently squashed commit (+ 1 commit for the merge itself and conflict resolution).
